cpp/betweenness_centrality_module/algorithm_online/betweenness_centrality_online.hpp

```diff
@@ -0,0 +1,255 @@
+  /// containing said node. The normalization constant is 2/((N-1)(N-2)) for undirected and 1/((N-1)(N-2)) for directed
+  /// graphs, with N being the number of graph nodes.
+  ///
+  ///@param node_bc_scores Graph nodes with associated betweenness centrality scores
+  ///@param graph_order Order (Nº of nodes) of the graph
+  ///
+  ///@return Normalized betweenness centrality scores
+  std::unordered_map<std::uint64_t, double> NormalizeBC(const std::unordered_map<std::uint64_t, double> &node_bc_scores,
+                                                        const std::uint64_t graph_order) const;
+
+  ///@brief Wrapper for the offline betweenness centrality algorithm that maps betweenness centrality scores to their
+  /// nodes’ Memgraph IDs.
+  ///
+  ///@param graph Current graph
+  ///@param threads Number of concurrent threads
+  void CallBrandesAlgorithm(const mg_graph::GraphView<> &graph, const std::uint64_t threads);
+
+  ///@brief Performs a breadth-first search from a given node with the Brandes’ algorithm.
+  ///
+  ///@param graph Graph traversed the BFS
+  ///@param start_id Start node ID
+  ///@param compensate_for_deleted_node If true, guarantees correct results for the DETACH_DELETE_NODE operation
+  ///
+  ///@return Nº of shortest paths to each node from the start node
+  /// Each node’s immediate predecessors on the shortest paths from the start node
+  /// IDs of nodes visited by the BFS, in reverse order
+  std::tuple<std::unordered_map<std::uint64_t, std::uint64_t>,
+             std::unordered_map<std::uint64_t, std::set<std::uint64_t>>, std::vector<std::uint64_t>>
+  BrandesBFS(const mg_graph::GraphView<> &graph, const std::uint64_t start_id,
+             const bool compensate_for_deleted_node = false) const;
+
+  ///@brief Returns the nodes and the articulation points of the biconnected component affected by the update.
+  ///
+  ///@param graph Graph containing affected edge (new graph if edge was created, old graph if edge was deleted)
+  ///@param updated_edge Created/deleted edge
+  ///
+  ///@return (nodes, articulation points) in the affected biconnected component
+  std::tuple<std::unordered_set<std::uint64_t>, std::unordered_set<std::uint64_t>> IsolateAffectedBCC(
+      const mg_graph::GraphView<> &graph, const std::pair<std::uint64_t, std::uint64_t> updated_edge) const;
+
+  ///@brief Computes lengths of shortest paths between a given node and all other nodes in the graph, restricted to
+  /// one biconnected component.
+  ///
+  ///@param graph Graph traversed by the algorithm
+  ///@param start_id Start node ID
+  ///@param bcc_nodes Nodes in the biconnected component
+  ///
+  ///@return {node ID, start->node distance} pairs
+  std::unordered_map<std::uint64_t, std::uint64_t> SSSPLengths(
+      const mg_graph::GraphView<> &graph, const std::uint64_t start_id,
+      const std::unordered_set<std::uint64_t> &bcc_nodes) const;
```

Comment:
Could this be encapsulated in one of the BFS methods to not repeat the process, or it would be too much work?

Reply:
Any of the BFS methods could be used for this, but at cost of making unnecessary data structures (e.g. predecessors, Ns of shortest paths) when the algorithm only needs the distances.